packages/blade/src/components/Drawer/Drawer.web.tsx

```diff
@@ -0,0 +1,225 @@
+        {isMounted ? (
+          <FloatingFocusManager context={context} initialFocus={initialFocusRef ?? closeButtonRef}>
+            <Box
+              position="fixed"
+              {...metaAttribute({
+                name: MetaConstants.Drawer,
+                testID,
+              })}
+              // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+              zIndex={zIndexState}
+            >
+              {showOverlay || stackingLevel === 2 ? (
+                <DrawerOverlay
+                  onClick={() => {
+                    onDismiss();
+                  }}
+                  className={isVisible ? SHOW_DRAWER : ''}
+                  lockScroll={true}
+                />
+              ) : null}
+              <AnimatedDrawerContainer
+                isVisible={isVisible}
+                isFirstDrawerInStack={isFirstDrawerInStack}
+                width={{ base: '90%', s: '375px', m: '420px' }}
+                {...makeAccessible({
+                  role: 'dialog',
+                  modal: true,
+                  label: accessibilityLabel,
+                })}
+                position="fixed"
+                top="spacing.0"
+                left="100%"
+                backgroundColor="popup.background.subtle"
+                elevation="highRaised"
+                height="100%"
+                display="flex"
+                flexDirection="column"
+                ref={refs.setFloating}
+                onKeyDown={(event) => {
+                  if (event?.key === 'Escape' || event?.code === 'Escape') {
+                    onDismiss();
+                  }
+                }}
+              >
+                {children}
+              </AnimatedDrawerContainer>
+            </Box>
+          </FloatingFocusManager>
+        ) : null}
+      </FloatingPortal>
+    </DrawerContext.Provider>
+  );
+};
+
+/**
+ * ### Drawer Component
+ * 
+ * A drawer is a panel that slides in mostly from right side of the screen over the existing content in the viewport. 
+ * It helps in providing additional details or context and can also be used to promote product features or new products.
+ * 
+ * ---
+ * 
+ * #### Usage
+ * 
+ * ```jsx
+  const MyDrawer = () => {
+    const [showDrawer, setShowDrawer] = React.useState(false);
+    return (
+      <Box>
+        <Button onClick={() => setShowDrawer(true)}>Open Drawer</Button>
+        <Drawer
+          isOpen={showDrawer}
+          onDismiss={() => setShowDrawer(false)}
+        >
+          <DrawerHeader title="Announcements" />
+          <DrawerBody>
+            <FTXAnnouncement />
+            <CatPictures />
+          </DrawerBody>
+        <Drawer>
+      </Box>
+    )
+  }
+ * ```
+ *
+ *  ---
+ *
+ * Checkout {@link https://blade.razorpay.com/?path=/docs/components-drawer Drawer Documentation}
+ * 
+ * 
+ */
+const Drawer = assignWithoutSideEffects(_Drawer, {
+  componentId: drawerComponentIds.Drawer,
+});
+
+export { Drawer };
```

Comment:
Export from `components/index.ts` as well

Reply:
did it in documentation PR